packages/ui/src/working-groups/hooks/useGroupActivities.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+
+import { useGetGroupEventsQuery } from '@/working-groups/queries'
+import {
+  asApplicationWithdrawnActivity,
+  asAppliedOnOpeningActivity,
+  asBudgetSpendingActivity,
+} from '@/working-groups/types/WorkingGroupActivity'
+
+export const useGroupActivities = (groupId: string) => {
+  const { loading, data } = useGetGroupEventsQuery({ variables: { group_eq: groupId } })
+  const activities = useMemo(
+    () =>
+      data
+        ? [
+            ...data.appliedOnOpeningEvents.map(asAppliedOnOpeningActivity),
+            ...data.applicationWithdrawnEvents.map(asApplicationWithdrawnActivity),
+            ...data.budgetSpendingEvents.map(asBudgetSpendingActivity),
+          ].sort((a, b) => b.createdAt.localeCompare(a.createdAt))
+        : [],
+    [data, loading]
+  )
+
+  return { isLoading: loading, activities }
+}
```

Comment:
That's an improvement as it allows to query for the actual events for the working group.

I'd rather see that we could have this returned from the API. However, If we'd get that from the API in the future it looks like the change would be limited to this hook and all the dependant code should stay the same.